src/ripple/app/tx/impl/SetHook.cpp

```diff
@@ -714,23 +724,9 @@ SetHook::preflight(PreflightContext const& ctx)
 
         allBlank = false;
 
-        for (auto const& hookSetElement : hookSetObj)
-        {
-            auto const& name = hookSetElement.getFName();
-
-            if (name != sfCreateCode && name != sfHookHash &&
-                name != sfHookNamespace && name != sfHookParameters &&
-                name != sfHookOn && name != sfHookGrants &&
-                name != sfHookApiVersion && name != sfFlags)
-            {
-                JLOG(ctx.j.trace())
-                    << "HookSet(" << hook::log::HOOK_INVALID_FIELD << ")["
-                    << HS_ACC()
-                    << "]: Malformed transaction: SetHook sfHook contains "
-                       "invalid field.";
-                return temMALFORMED;
-            }
-        }
```

Comment:
did you remove this code because it's checked by the template system already?

Reply:
Yes, Also, it is not possible to check optional fields that are not specified in the transaction within this process. (This is because all fields of the Hook object, including optional fields, are included)

I think a similar process was used in sFMemos as well.

Reply:
It's just a whitelist of allowable fields inside the inner hookset obj.

Is there currently a unit test for this? If not can you add one (illegal fields inside hookset obj) and ensure the preflight is still temMALFORMED in the revised code?

Reply:
Currently, as a result of the template check, this is an invalidTransaction error, not a temMALFORMED error.

The error occurs at the time of the submit command.


I can add tests to check for invalidTransaction errors, but generally such tests are not present in other test cases.

Reply:
I think from a defensive programming standpoint there's no harm in leaving this code in? Just add sfCanEmit to the whitelist?

Reply:
added